.github/workflows/ci.yml

```diff
@@ -39,17 +39,17 @@ jobs:
       - run: npm ci
       - run: npm run codegen:apollo
       - run: npm run lint
-      # - run: npm test
+      - run: npm test
       - run: npm run build
 
-  # coverage:
-  #   runs-on: ubuntu-latest
-  #   steps:
-  #     - uses: actions/checkout@v2
-  #     - uses: actions/setup-node@v2
-  #     - run: npm ci
-  #     - uses: paambaati/codeclimate-action@v2.7.5
-  #       env:
-  #         CC_TEST_REPORTER_ID: ${{ secrets.CC_TEST_REPORTER_ID }}
-  #       with:
-  #         coverageCommand: npm test
+  coverage:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+      - run: npm ci
+      - uses: paambaati/codeclimate-action@v2.7.1 # using 2.7.1 for issue: https://github.com/paambaati/codeclimate-action/issues/316
+        env:
+          CC_TEST_REPORTER_ID: ${{ secrets.CC_TEST_REPORTER_ID }}
+        with:
+          coverageCommand: npm test
```

Comment:
so this effectively makes `coverage` job run same things as current `test` job, as most likely we will have to pregenerate things here too for test run to work.

Better approach to limit duplicated runs and reduce test time would be to upload/download artifacts from the `test` job, as Jest generates coverage files by default whenever we run `npm test`.

Reply:
what about having this step inside the `test` job? Something like:

`
test:
    runs-on: ${{ matrix.os }}

    strategy:
      fail-fast: false
      matrix:
        os: [ubuntu-latest, macos-latest, windows-latest]
        node: ['16', '14']

    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v2
        with:
          node-version: ${{ matrix.node }}

      - name: Cache node_modules
        uses: actions/cache@v2
        env:
          cache-name: cache-node-modules
        with:
          path: ~/.npm
          key: ${{ runner.os }}-${{ matrix.node }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
          restore-keys: ${{ runner.os }}-${{ matrix.node }}-build-${{ env.cache-name }}-

      - run: npm ci
      - run: npm run build

      - name: Code climate action
      - uses: paambaati/codeclimate-action@v2.7.1 # using 2.7.1 for issue: https://github.com/paambaati/codeclimate-action/issues/316
        env:
          CC_TEST_REPORTER_ID: ${{ secrets.CC_TEST_REPORTER_ID }}
        with:
          coverageCommand: npm test
`

We can also run the `npm test` outside `coverageCommand` as documented: ["When you've already generated the coverage report in a previous step and wish to just upload the coverage data to Code Climate, you can leave out the coverageCommand option."](https://github.com/paambaati/codeclimate-action)

PS: sorry for posting the code here, but i want to avoid stressing too much on this PR with commits. I might need to setup [Art](https://github.com/nektos/act) to test these actions locally to avoid having to push every time.

Reply:
yup, that too. Basically the jobs don't share a common file space so while we can generate coverage whenever we run npm test, the output has to uploaded/downloaded as artifacts as this is the only way for multiple jobs sharing the same files, which then is why `coverage` should only run after `test` job ran, hence only when it was successful.